rinja_derive/src/input.rs

```diff
@@ -422,6 +427,127 @@ impl TemplateArgs {
+    Some((source, span))
+}
+
+#[cfg(feature = "code-in-doc")]
+fn collect_comment_blocks(ast: &syn::DeriveInput) -> Option<(Option<Span>, String)> {
+    let mut span: Option<Span> = None;
+    let mut assign_span = |kv: &syn::MetaNameValue| {
+        // FIXME: uncomment once <https://github.com/rust-lang/rust/issues/54725> is stable
+        // let new_span = kv.path.span();
+        // span = Some(match span {
+        //     Some(cur_span) => cur_span.join(new_span).unwrap_or(cur_span),
+        //     None => new_span,
+        // });
+
+        if span.is_none() {
+            span = Some(kv.path.span());
+        }
+    };
+
+    let mut source = String::new();
+    for a in &ast.attrs {
+        // is a comment?
+        let syn::Meta::NameValue(kv) = &a.meta else {
+            continue;
+        };
+        if !kv.path.is_ident("doc") {
+            continue;
+        }
+
+        // is an understood comment, e.g. not `#[doc = inline_str(…)]`
+        let mut value = &kv.value;
+        let value = loop {
+            match value {
+                syn::Expr::Lit(lit) => break lit,
+                syn::Expr::Group(group) => value = &group.expr,
+                _ => continue,
+            }
+        };
+        let syn::Lit::Str(value) = &value.lit else {
+            continue;
+        };
+
+        assign_span(kv);
+        source.push_str(value.value().as_str());
+        source.push('\n');
+    }
+    if source.is_empty() {
+        return None;
+    }
+
+    Some((span, source))
+}
+
+#[cfg(feature = "code-in-doc")]
+fn strip_common_ws_prefix(source: String) -> Option<String> {
+    let mut common_prefix_iter = source
+        .lines()
+        .filter_map(|s| Some(&s[..s.find(|c: char| !c.is_ascii_whitespace())?]));
+    let mut common_prefix = common_prefix_iter.next().unwrap_or_default();
+    for p in common_prefix_iter {
+        if common_prefix.is_empty() {
+            break;
+        }
+        let ((pos, _), _) = common_prefix
+            .char_indices()
+            .zip(p.char_indices())
+            .take_while(|(l, r)| l == r)
+            .last()
+            .unwrap_or_default();
+        common_prefix = &common_prefix[..pos];
+    }
+    if common_prefix.is_empty() {
+        return Some(source);
+    }
+
+    Some(
+        source
+            .lines()
+            .flat_map(|s| [s.get(common_prefix.len()..).unwrap_or_default(), "\n"])
+            .collect(),
+    )
+}
+
+#[cfg(feature = "code-in-doc")]
+fn collect_rinja_code_blocks(source: String) -> Option<Source> {
+    use pulldown_cmark::{CodeBlockKind, Event, Parser, Tag, TagEnd};
+
+    let mut tmpl_source = String::new();
+    let mut in_rinja_code = false;
+    let mut had_rinja_code = false;
+    for e in Parser::new(&source) {
+        match (in_rinja_code, e) {
+            (false, Event::Start(Tag::CodeBlock(CodeBlockKind::Fenced(s)))) => {
+                if s.split(",").any(|s| JINJA_EXTENSIONS.contains(&s)) {
```

Comment:
In this case I think it should be:

```suggestion
                if s.split(",").any(|s| JINJA_EXTENSIONS.contains(&s)) {
```

The reason is because code block attributes are actually a bit complicated: https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#custom-css-classes-for-code-blocks

Reply:
... Nevermind, it's exactly what you're doing. End of week urg.